From a72a5fc53e5486c0c699bfab59476f3397d76dc3 Mon Sep 17 00:00:00 2001
From: Nuri Jung <jnooree@snu.ac.kr>
Date: Fri, 27 Jun 2025 11:38:05 +0900
Subject: [PATCH] stage-2

---
 src/main.py | 24 +++++++++++++++++-------
 1 file changed, 17 insertions(+), 7 deletions(-)

diff --git a/src/main.py b/src/main.py
index 00fa75e..0ad9b4f 100644
--- a/src/main.py
+++ b/src/main.py
@@ -7,23 +7,33 @@ import pdbline
 
 
 def read_pdb(pdb: Path):
-    coords: list[np.ndarray] = []
+    models: list[np.ndarray] = []
 
     with open(pdb) as f:
+        model: list[np.ndarray] = []
+
         for line in f:
             record = pdbline.record(line)
-            if record == "ATOM" or record == "HETATM":
+            if record == "MODEL":
+                if model:
+                    models.append(np.stack(model))
+                model = []
+            elif record == "ATOM" or record == "HETATM":
                 if pdbline.res_seq(line) % 5 == 0:
-                    coords.append(pdbline.coordinates(line))
+                    model.append(pdbline.coordinates(line))
 
-    return np.stack(coords)
+        if model:
+            models.append(np.stack(model))
+
+    return models
 
 
 def main():
     pdb_file = Path(sys.argv[1])
-    coords = read_pdb(pdb_file)
-    com = np.mean(coords, axis=0)
-    print(f"{com = }")
+    models = read_pdb(pdb_file)
+    for model, coords in enumerate(models):
+        com = np.mean(coords, axis=0)
+        print(f"{model = }: {com = }")
 
 
 if __name__ == "__main__":
-- 
2.50.0

